oberon07/CompleteUnscientificBenchmark.obn

```diff
@@ -96,7 +87,6 @@ VAR
 BEGIN
    Split3(t.root, lower, equal, greater, val);
    t.root := Merge(lower, greater);
-   DeleteNode(equal);
```

Comment:
Won't it leak memory?

Reply:
Oberon is a garbage-collected language, so it shouldn't. I think I originally had `DeleteNode()` in there because I copied from Modula-2, which is not garbage-collected, so I just never thought about it. In addition, while trying to figure out how to make the Eiffel code more efficient (I've given up on that now) I noticed that, once you get to this point, `equal` has had both its leaves set to `NIL`, so there shouldn't be any unfortunate references that could confuse a collector.

Reply:
Having replied, I'm not sure if I should click "Resolve conversation" or wait for you to review it, so I'll leave it be for now.